### ISSUE TITLE
OnClick not working with jquery-validation-unobtrusive

### ISSUE THREAD

**Issue by mitrovicsinisaa** [aspnet/jquery-validation-unobtrusive#40]
Hello,

I have specific problem, and I'm not sure if this is due jquery-validation-unobtrusive or jquery-validation itself. I have ASP MVC 5 project with validation. Form is submitted via AJAX. Unobtrusive work on focus event of other element as far as I could see, and it call onError and onSuccess methods of unobtrusive. Problem I have is that if I have error on input field, and I insert correct value, and wright after that I click on submit button unobtrusive will take over and fire onError and onSuccess methods and submit button onClick event will not be called. Is there any instruction to fix this?

Best regards,
Sinisa Mitrovic

**Comment by Eilon** [comment]
@ryanbrandenburg can you have a look? This is with ASP.NET MVC 5, but it's pretty much the same client side code as the new stuff we have, so we could try to repro with that first and see if it's still an issue.

**Comment by mitrovicsinisaa** [comment]
After additional testing issue can be reproduced easily. Unobtrusive fire onError and onSuccess methods after field lose focus. So if I clock anywhere but on submit button, it will work as expected. 
Problem is direct clicking on submit, in that case Unobtrusive fire events, and escape click event of button.

**Comment by ryanbrandenburg** [comment]
@mitrovicsinisaa I was unable to reproduce this. I used [this](https://github.com/aspnet/Mvc/tree/rybrande/validation/test/WebSites/BasicWebSite) site on that branch to try to replicate what you described. Does the cshtml for RemoteAttribute_Home/Create accuratly reflect what you're trying to do? If not then let me know where I went wrong. If it does, then there's likely a problem in either your app or an older version of jquery-validation-unobtrusive.
